client/collection.go

```diff
@@ -16,24 +16,34 @@ import (
 	"github.com/sourcenetwork/defradb/datastore"
 )
 
+// CollectionDefinition contains the metadata defining what a Collection is.
+type CollectionDefinition struct {
+	// Description returns the CollectionDescription of this Collection.
+	Description CollectionDescription `json:"description"`
+	// Schema returns the SchemaDescription used to define this Collection.
+	Schema SchemaDescription `json:"schema"`
+}
+
 // Collection represents a defradb collection.
 //
 // A Collection is mostly analogous to a SQL table, however a collection is specific to its
 // host, and many collections may share the same schema.
 //
 // Many functions on this object will interact with the underlying datastores.
 type Collection interface {
-	// Description returns the CollectionDescription of this Collection.
-	Description() CollectionDescription
 	// Name returns the name of this collection.
 	Name() string
-	// Schema returns the SchemaDescription used to define this Collection.
-	Schema() SchemaDescription
 	// ID returns the ID of this Collection.
 	ID() uint32
 	// SchemaID returns the ID of the Schema used to define this Collection.
 	SchemaID() string
 
+	Definition() CollectionDefinition
```

Comment:
suggestion: This should probably have documentation.

Reply:
will add

- [ ] add doc